src/Core/StorageManager.php

```diff
@@ -61,93 +62,115 @@ class StorageManager
-	private $httpRequest;
 
-	/** @var Storage\IStorage */
+	/** @var Storage\IWritableStorage */
 	private $currentBackend;
 
 	/**
 	 * @param Database        $dba
 	 * @param IConfig         $config
 	 * @param LoggerInterface $logger
 	 * @param L10n            $l10n
+	 *
+	 * @throws Storage\InvalidClassStorageException in case the active backend class is invalid
+	 * @throws Storage\StorageException in case of unexpected errors during the active backend class loading
 	 */
-	public function __construct(Database $dba, IConfig $config, LoggerInterface $logger, L10n $l10n, IHTTPRequest $httpRequest)
+	public function __construct(Database $dba, IConfig $config, LoggerInterface $logger, L10n $l10n)
 	{
-		$this->dba         = $dba;
-		$this->config      = $config;
-		$this->logger      = $logger;
-		$this->l10n        = $l10n;
-		$this->httpRequest = $httpRequest;
-		$this->backends = $config->get('storage', 'backends', self::DEFAULT_BACKENDS);
+		$this->dba      = $dba;
+		$this->config   = $config;
+		$this->logger   = $logger;
+		$this->l10n     = $l10n;
+		$this->validBackends = $config->get('storage', 'backends', self::DEFAULT_BACKENDS);
 
-		$currentName = $this->config->get('storage', 'name', '');
+		$currentName = $this->config->get('storage', 'name');
 
 		// you can only use user backends as a "default" backend, so the second parameter is true
-		$this->currentBackend = $this->getByName($currentName, true);
+		$this->currentBackend = $this->getWritableStorageByName($currentName);
 	}
 
 	/**
 	 * Return current storage backend class
 	 *
-	 * @return Storage\IStorage|null
+	 * @return Storage\IWritableStorage
 	 */
 	public function getBackend()
 	{
 		return $this->currentBackend;
 	}
 
 	/**
-	 * Return storage backend class by registered name
+	 * Returns a writable storage backend class by registered name
 	 *
-	 * @param string|null $name            Backend name
-	 * @param boolean     $onlyUserBackend True, if just user specific instances should be returrned (e.g. not SystemResource)
+	 * @param string $name Backend name
 	 *
-	 * @return Storage\IStorage|null null if no backend registered at $name
+	 * @return Storage\IWritableStorage
 	 *
-	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws Storage\InvalidClassStorageException in case there's no backend class for the name
+	 * @throws Storage\StorageException in case of an unexpected failure during the hook call
 	 */
-	public function getByName(string $name = null, $onlyUserBackend = false)
+	public function getWritableStorageByName(string $name): Storage\IWritableStorage
 	{
-		// @todo 2020.09 Remove this call after 2 releases
-		$name = $this->checkLegacyBackend($name);
+		$storage = $this->getByName($name, $this->validBackends);
+		if ($storage instanceof Storage\IWritableStorage) {
+			return $storage;
+		} else {
+			throw new Storage\InvalidClassStorageException(sprintf('Backend %s is not writable', $name));
+		}
```

Comment:
Nitpicking, this structure can get rid of the `else`.
```suggestion
		if (!$storage instanceof Storage\IWritableStorage) {
			throw new Storage\InvalidClassStorageException(sprintf('Backend %s is not writable', $name));
		}
		
		return $storage;
```